ajax/loadfile.php

```diff
@@ -36,19 +36,26 @@
 	$writeable = \OC\Files\Filesystem::isUpdatable($path);
 	$mime = \OC\Files\Filesystem::getMimeType($path);
 	$mtime = \OC\Files\Filesystem::filemtime($path);
-	$filecontents = \OC\Files\Filesystem::file_get_contents($path);
-	$encoding = mb_detect_encoding($filecontents."a", "UTF-8, WINDOWS-1252, ISO-8859-15, ISO-8859-1, ASCII", true);
-	if ($encoding == "") {
-		// set default encoding if it couldn't be detected
-		$encoding = 'ISO-8859-15';
+	$maxsize = \OCP\Config::getSystemValue("max_size_text_editor", 4194304); // default of 4MB
+	$size = \OC\Files\Filesystem::filesize($path);
+	if($size > $maxsize) {
+		OCP\JSON::error(array('data' => array( 'message' => "Max size for opening files is $maxsize.")));
+	} else {
+		$filecontents = \OC\Files\Filesystem::file_get_contents($path);
+		$encoding = mb_detect_encoding($filecontents."a", "UTF-8, WINDOWS-1252, ISO-8859-15, ISO-8859-1, ASCII", true);
```

Comment:
I'd prefer if we would just read the first 500/1000/whatever characters and guess the encoding of that one. - No need to do this over the whole file from my PoV.